### ISSUE TITLE
How to take snapshot before delete an index via ILM?

### ISSUE THREAD

**Issue by hatharom** [elastic/elasticsearch#59418]
I dont know whether it is a question, a bug or a feature request.

All I want is to delete indices say every month after a snapshot is taken.

I have an ILM definition like this:

```
{
  "policy": {
    "phases": {
      "delete": {
        "actions": {
          "wait_for_snapshot" : {
            "policy": "slm-policy-name"
          }
        }
      }
    }
  }
}
```
The problem is that it wont trigger the slm but waiting for it.

I can solve it currently by  for example an slm trigger at every night. But this can bring an 24h wait period in the worst case(If the index creation happened just after the slm snapshot trigger time, then the deletion will be triggered after the snapshot is taken so it waits almost 24h for the next snapshot time).
On the other hand I don't want to take snapshots all the time and for all the indices but only for the specific index thats will be deleted and only at that time(ie monthly snapshots). 
How to do this?

So in summary I am looking for an "do_snapshot_then_delete" solution instead of "wait_for_snapshot"

**Comment by astefan** [comment]
@hatharom Thanks very much for your interest in Elasticsearch and ILM.
This appears to be a user question, and we'd like to direct these kinds of things to the [forums](https://discuss.elastic.co/). If you can stop by there, we'd appreciate it. This allows us to use GitHub for verified bug reports, feature requests, and pull requests.

There's an active community in the [forums](https://discuss.elastic.co/) that should be able to help get an answer to your question. As such, I hope you don't mind that I close this.
